src/auth/auth-manager.ts

```diff
@@ -1,62 +1,119 @@
+import {
+  AccountInfo,
+  InteractionRequiredAuthError,
+  PublicClientApplication,
+} from '@azure/msal-browser';
 import { jwtDecode } from 'jwt-decode';
+import { msalConfig, SCOPES } from '../api/constants';
 import { TokenType } from '../enum';
-import {
-  AccessTokenDecodeType,
-  AuthenticationResponseType,
-  RequestUser,
-} from '../types';
-import { login, refreshSession } from '../api/api';
-
-export const logoutUser: Function = () => {
-  Object.keys(TokenType).forEach((token) =>
-    window.localStorage.removeItem(token)
-  );
-};
+import { TokenDecodeType, AuthenticationResponseType } from '../types';
 
-export const saveUserDataLocalStorage: Function = ({
-  id_token,
-  access_token,
-  refresh_token,
-}: AuthenticationResponseType) => {
-  if (refresh_token) {
-    window.localStorage.setItem(TokenType.refresh, refresh_token);
-  }
-  window.localStorage.setItem(TokenType.bearer, id_token);
-  window.localStorage.setItem(TokenType.access, access_token);
-};
+export const AuthManager = () => {
+  let msalInst: PublicClientApplication | null = null;
 
-export const getToken: Function = (typeToken: TokenType) => {
-  return window.localStorage.getItem(typeToken);
-};
+  const init: Function = async () => {
+    msalInst = new PublicClientApplication(msalConfig);
+    await msalInst.initialize();
+  };
 
-export const isValidAccessToken: Function = () => {
-  const token = getToken(TokenType.access);
-  if (!token) {
-    return false;
-  }
-  const accessTokenDecode: AccessTokenDecodeType = jwtDecode(token);
-  const expDate = new Date(accessTokenDecode?.exp * 1000);
-  return expDate > new Date();
-};
+  const saveUserDataLocalStorage: Function = (
+    token: AuthenticationResponseType
+  ) => {
+    window.localStorage.setItem(TokenType.idToken, token?.idToken);
+    window.localStorage.setItem(TokenType.accessToken, token?.accessToken);
+  };
+
+  const getAccessToken: Function = () => {
+    return window.localStorage.getItem(TokenType.accessToken);
+  };
+
+  const getIdToken: Function = () => {
+    return window.localStorage.getItem(TokenType.idToken);
+  };
+
+  const getCurrentAccount: Function = async () => {
```

Comment:
Usually, we would expect that a method like that would be called very often in the app, so it is better to make it simple enough, so that we do not decode the token each time.

Reply:
fixed